pkg/util/clusterdata/ingress_profiles_task.go

```diff
@@ -0,0 +1,129 @@
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+func newIngressProfileEnricherTask(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftCluster) (enricherTask, error) {
+	opClient, err := operatorclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	kubClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ingressProfileEnricherTask{
+		log:       log,
+		opClient:  opClient,
+		kubClient: kubClient,
+		oc:        oc,
+	}, nil
+}
+
+type ingressProfileEnricherTask struct {
+	log       *logrus.Entry
+	opClient  operatorclient.Interface
+	kubClient kubernetes.Interface
+	oc        *api.OpenShiftCluster
+}
+
+func (ef *ingressProfileEnricherTask) FetchData(ctx context.Context, callbacks chan<- func(), errs chan<- error) {
+
+	//List IngressControllers from  openshift-ingress-operator namespace
+	//Each IngressController will be the basis for an IngressProfile with the below mapping:
+	//     IngressController.Name -> IngressProfile.Name
+	//     IngressController.spec.endpointPublishingStrategy.loadBalancer.scope -> IngressProfile.Visibility
+	//             Internal -> Private
+	//             External -> Public
+	ingressControllers, err := ef.opClient.OperatorV1().IngressControllers("openshift-ingress-operator").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		ef.log.Error(err)
+		errs <- err
+		return
+	}
+
+	//List Services from openshift-ingress namespace
+	//Among those Services, look for the ones of type LoadBalancer, with name "router-xxx". The matching will be done with
+	//IngressController name based on this xxx.
+	//Service IP will be taken from the candidate and added to the corresponding IngressProfile
+	services, err := ef.kubClient.CoreV1().Services("openshift-ingress").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		ef.log.Error(err)
+		errs <- err
+		return
+	}
+
+	routerIPs := make(map[string]string)
+	for _, service := range services.Items {
+		svcName := service.ObjectMeta.Name
+		if len(svcName) < 8 || !strings.HasPrefix(svcName, "router-") || len(service.Status.LoadBalancer.Ingress) == 0 {
+			continue
+		}
+		matchingICName := svcName[7:] //router-xxx
+		routerIPs[matchingICName] = service.Status.LoadBalancer.Ingress[0].IP
+	}
+
+	//Reconcile IngressController and corresponding router service
+	ingressProfiles := make([]api.IngressProfile, len(ingressControllers.Items))
+	for i, ingressController := range ingressControllers.Items {
+		ingressProfiles[i] = api.IngressProfile{
+			Name: ingressController.ObjectMeta.Name,
+		}
+
+		// set IP
+		if IP, ok := routerIPs[ingressProfiles[i].Name]; ok {
```

Comment:
`s/IP/ip`

Reply:
I somehow think if make array stronger in consistency above you would be able to skip these checks and just assign directly. 
This function is big complicated to read now :/